File: MP3-transcoding--master/transcoding(MP3)/ViewController.m
//
//  ViewController.m
//  transcoding(MP3)
//
//  Created by xindong on 16/11/12.
//  Copyright © 2016年 xindong. All rights reserved.
//

#import "ViewController.h"
#import "XDRecorder.h"

@interface ViewController ()<UpdateValueDelegate>
@property (weak, nonatomic) IBOutlet UILabel *displayLabel;
@property (weak, nonatomic) IBOutlet UIButton *recordButton;
@property (weak, nonatomic) IBOutlet UIButton *transcodingButton;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    self.displayLabel.text = @"准备录音";
    [[XDRecorder sharedRecorder] setDelegate:self];
    
}

#pragma mark - UpdateValueDelegate
- (void)updateTextValue:(NSString *)text {
    self.displayLabel.text = text;
}

- (IBAction)clickedRecordButton:(UIButton *)sender {
    sender.selected = !sender.selected;
    if (sender.selected) [XDRecorder starRecordingWithAutoTranscoding:NO];
    else [XDRecorder stopRecording];
}

- (IBAction)clickedTranscodingButton:(UIButton *)sender {
    [XDRecorder transcodingToMP3];
}



- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: MagiAlert/MagiAlert/ViewController.swift
//
//  ViewController.swift
//  MagiAlert
//
//  Created by 安然 on 2017/4/1.
//  Copyright © 2017年 安然. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    @IBAction func tanChu(_ sender: UIButton) {
        
        MagiAlert().showAlert("选择支付方式",
                              price: "¥1.00",
                              content: "可用余额12.5元",
                              buttonTitle: "取消",
                              otherButtonTitle: "确认支付") { (sender) -> Void in
                                print("执行了确认")
        }
    }


}

